File: app/src/sample/java/com/mycompany/myapp/app/SSLDevelopmentHelper.java
package com.mycompany.myapp.app;

import java.security.cert.CertificateException;
import java.security.cert.X509Certificate;

import javax.net.ssl.HostnameVerifier;
import javax.net.ssl.SSLContext;
import javax.net.ssl.SSLSession;
import javax.net.ssl.TrustManager;
import javax.net.ssl.X509TrustManager;

public class SSLDevelopmentHelper {

    public static HostnameVerifier buildTrustAllHostnameVerifer() {
        return new HostnameVerifier() {
            public boolean verify(String string, SSLSession ssls) {
                return true;
            }
        };
    }

    public static SSLContext buildTrustAllSSLContext() {
        // Create a trust manager that does not validate certificate chains to
        // avoid "Trust anchor for certification path not found" exception
        TrustManager[] trustAllCerts = new TrustManager[]{buildTrustAllTrustManager()};

        try {
            SSLContext sc = SSLContext.getInstance("TLS");  //NON-NLS
            sc.init(null, trustAllCerts, new java.security.SecureRandom());
            return sc;
        } catch (Exception ex) {
            throw new RuntimeException(ex);
        }
    }

    public static X509TrustManager buildTrustAllTrustManager() {
        return new X509TrustManager() {
            public X509Certificate[] getAcceptedIssuers() {
                return new X509Certificate[]{};
            }

            public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
            }

            public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {
            }
        };
    }
}

File: app/src/dev/java/com/atomicrobot/carbon/app/SSLDevelopmentHelper.kt
package com.atomicrobot.carbon.app

import android.annotation.SuppressLint
import okhttp3.OkHttpClient
import java.security.SecureRandom
import java.security.cert.CertificateException
import java.security.cert.X509Certificate
import javax.net.ssl.HostnameVerifier
import javax.net.ssl.SSLContext
import javax.net.ssl.TrustManager
import javax.net.ssl.X509TrustManager

object SSLDevelopmentHelper {

    fun applyTrustAllSettings(builder: OkHttpClient.Builder): OkHttpClient.Builder {
        return builder.apply {
            hostnameVerifier(buildTrustAllHostnameVerifier())

            val trustManager = buildTrustAllTrustManager()
            val sslContext = buildTrustAllSSLContext()
            sslContext.init(null, arrayOf(trustManager), SecureRandom())
            sslSocketFactory(sslContext.socketFactory, trustManager)
        }
    }

    private fun buildTrustAllHostnameVerifier(): HostnameVerifier {
        return HostnameVerifier { _, _ -> true }
    }

    private fun buildTrustAllSSLContext(): SSLContext {
        // Create a trust manager that does not validate certificate chains to
        // avoid "Trust anchor for certification path not found" exception
        val trustAllCerts = arrayOf<TrustManager>(buildTrustAllTrustManager())

        try {
            val sc = SSLContext.getInstance("TLS") // NON-NLS
            sc.init(null, trustAllCerts, SecureRandom())
            return sc
        } catch (ex: Exception) {
            throw RuntimeException(ex)
        }
    }

    private fun buildTrustAllTrustManager(): X509TrustManager {
        return object : X509TrustManager {
            override fun getAcceptedIssuers(): Array<X509Certificate> {
                return arrayOf()
            }

            @SuppressLint("TrustAllX509TrustManager")
            @Throws(CertificateException::class)
            override fun checkClientTrusted(chain: Array<X509Certificate>, authType: String) {
            }

            @SuppressLint("TrustAllX509TrustManager")
            @Throws(CertificateException::class)
            override fun checkServerTrusted(chain: Array<X509Certificate>, authType: String) {
            }
        }
    }
}
